helix-term/src/ui/editor.rs

```diff
@@ -856,6 +858,93 @@ impl Component for EditorView {
                 doc.set_selection(view.id, selection);
                 EventResult::Consumed(None)
             }
+
+            Event::Mouse(MouseEvent {
+                kind: MouseEventKind::Up(MouseButton::Left),
+                ..
+            }) => {
+                if !self.middle_click_paste {
+                    return EventResult::Ignored;
+                }
+
+                let (view, doc) = current!(cx.editor);
+                let range = doc.selection(view.id).primary();
+
+                if range.to() - range.from() <= 1 {
+                    return EventResult::Ignored;
+                }
+
+                let mut cxt = commands::Context {
+                    selected_register: helix_view::RegisterSelection::default(),
+                    editor: &mut cx.editor,
+                    count: None,
+                    callback: None,
+                    on_next_key_callback: None,
+                    jobs: cx.jobs,
+                };
+
+                commands::Command::yank_main_selection_to_primary_clipboard.execute(&mut cxt);
+
+                EventResult::Consumed(None)
+            }
+
+            Event::Mouse(MouseEvent {
+                kind: MouseEventKind::Up(MouseButton::Middle),
+                row,
+                column,
+                modifiers,
+                ..
+            }) => {
+                if !self.middle_click_paste {
+                    return EventResult::Ignored;
+                }
+
+                let editor = &mut cx.editor;
```

Comment:
Why do we need this?

Reply:
It's needed later.

Reply:
Fixed (https://github.com/helix-editor/helix/pull/548#discussion_r683356890)